azurerm/internal/services/network/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -510,23 +510,19 @@ func getArmVirtualNetworkGatewayConnectionProperties(d *schema.ResourceData) (*n
 
 	if props.ConnectionType == network.ExpressRoute {
 		if props.Peer == nil || props.Peer.ID == nil {
-			return nil, fmt.Errorf("`express_route_circuit_id` must be specified when `type` is set to `ExpressRoute")
+			return nil, fmt.Errorf("`express_route_circuit_id` must be specified when `type` is set to `ExpressRoute`")
 		}
 	}
 
 	if props.ConnectionType == network.IPsec {
 		if props.LocalNetworkGateway2 == nil || props.LocalNetworkGateway2.ID == nil {
-			return nil, fmt.Errorf("`local_network_gateway_id` and `shared_key` must be specified when `type` is set to `IPsec")
-		}
-
-		if props.SharedKey == nil {
```

Comment:
can we add an acceptance test to ensure this use-case is supported through the API? Unfortunately ARM Templates can behave differently to the API's themselves - so an acceptance test should confirm this

Reply:
Fixed